Standards/scs-0217-v1-cluster-hardening.md

```diff
@@ -0,0 +1,475 @@
+For more information on high-availability, look into the [Kubernetes Node Distribution and Availability Standard](scs-0214-v1-k8s-node-distribution.md).
+It would also be possible to use these etcd instances in order to select specific instances
+that aren't the current etcd leader for interaction with different components (e.g. Calico), since access to the primary etcd instance could be considered dangerous, because the full keyspace could be viewed without further restrictions (see [here](https://cheatsheetseries.owasp.org/cheatsheets/Kubernetes_Security_Cheat_Sheet.html#limiting-access-to-the-primary-etcd-instance) or [here](https://docs.tigera.io/calico/latest/reference/etcd-rbac/kubernetes-advanced)).
+This approach should still be paired with [etcd ACL](#acl-restrictions) to better restrict access.
+
+#### etcd isolation
+
+The etcd database should at best be isolated from the rest of a Kubernetes cluster.
+Access should only be granted to components that need it, which is in most cases mainly (or only)
+the API server. Best practice would be to host etcd on machines separate from the Kubernetes cluster
+and block access from machines or networks that don't need access with specific firewall rules.
+In most cases, only the API server machines should need access to etcd on ports 2379-2380.
+
+#### ACL restrictions
+
+etcd implements access control lists (ACL) and authentication since version 2.1 [1][etcd-auth].
+etcd provides users and roles; users gain permissions through roles. When authentication is enabled,
+each request to etcd requires authentication and the transaction is only allowed, if the user has the correct access rights.
+etcd can also be launched with `--client-cert-auth=true`, which enables authentication via
+the Common Name (CN) field of a client TLS certificate without a password.
+This option enables Kubernetes components to authenticate as a user without providing a password,
+which is neither possible for Kubernetes components nor planned in future releases.
+This method is recommended in order to implement ACL for different Kubernetes components and
+not give the Kubernetes API full root access to the etcd instance; instead, a separate user can be created.
+
+#### TLS communication
+
+etcd should use TLS for peer- and cluster-communication, so that traffic between different peered etcd instances as well
+as the communication with the Kubernetes cluster can be secured.
+etcd provides options for all these scenarios, including `--peer-key-file=peer.key` and `--peer-cert-file=peer.cert`
+for securing peer communication and the flags `--key-file=k8sclient.key` and `--cert-file=k8sclient.cert` for securing
+client communication (and therefore cluster communication).
+Additionally, HTTPS should be used as the URL schema.
+It is also possibly to use a separate CA for the etcd in order to separate and better control access through client
```

Comment:
```suggestion
It is also possible to use a separate CA for the etcd in order to separate and better control access through client
```

Reply:
Updated in a2adca5b18c82097388b59a88dc665c0d2b72ab4